packages/tools/src/launchEditor.ts

```diff
@@ -96,27 +107,56 @@ function getArgumentsForLineNumber(
   }
 }
 
+function getArgumentsForFileName(
+  editor: string,
+  fileName: string,
+  workspace: any,
+) {
+  switch (path.basename(editor)) {
+    case 'code':
+      return addWorkspaceToArgumentsIfExists([fileName], workspace);
+    case 'devenv':
+      return ['/EDIT', fileName];
+    // Every other editor just takes the filename as an argument
+    default:
+      return [fileName];
+  }
+}
+
 function guessEditor() {
   // Explicit config always wins
   if (process.env.REACT_EDITOR) {
     return shellQuote.parse(process.env.REACT_EDITOR);
   }
 
-  // Using `ps x` on OSX we can find out which editor is currently running.
-  // Potentially we could use similar technique for Windows and Linux
-  if (process.platform === 'darwin') {
-    try {
+  try {
+    // Using `ps x` on OSX we can find out which editor is currently running.
+    // Potentially we could use similar technique for Linux
+    if (process.platform === 'darwin') {
       const output = execSync('ps x').toString();
       const processNames = Object.keys(COMMON_EDITORS);
-      for (let i = 0; i < processNames.length; i++) {
-        const processName = processNames[i];
+      for (const processName of processNames) {
         if (output.indexOf(processName) !== -1) {
           return [COMMON_EDITORS[processName]];
         }
       }
-    } catch (error) {
-      // Ignore...
+    } else if (process.platform === 'win32') {
+      const output = execSync(
+        'tasklist /NH /FO CSV /FI "SESSIONNAME ne Services"',
+      ).toString();
+
+      const runningProcesses = output
+        .split('\n')
+        .map(line => line.replace(/^"|".*\r$/gm, ''));
+      const processNames = Object.keys(COMMON_WINDOWS_EDITORS);
+      for (const processName of processNames) {
+        if (runningProcesses.includes(processName)) {
+          return [COMMON_WINDOWS_EDITORS[processName]];
+        }
+      }
```

Comment:
makes sense to reuse this logic from https://github.com/react-native-community/cli/pull/1222 maybe?